src/model/time_entry.cc

```diff
@@ -347,14 +339,7 @@ void TimeEntry::SetProjectGUID(const std::string &value, bool userModified) {
 }
 
 const std::string TimeEntry::Tags() const {
-    std::stringstream ss;
-    for (auto it = TagNames->begin(); it != TagNames->end(); ++it) {
-        if (it != TagNames->begin()) {
-            ss << kTagSeparator;
-        }
-        ss << *it;
-    }
-    return ss.str();
+    return std::string(TagsVectorToString(TagNames()));
```

Comment:
This results in copying the string in this case but for the case with tags in the database it avoids one copy so it's tradeoff of having a copy here or there...

Reply:
what if `TagsVectorToString` simply returns the result of `.str()` and we use `bind` instead of `useRef`?
so that would be the same number of copies in the database code and one less copy here if I'm not mistaken.

Reply:
Hmm I probably need to finish my coffee because I have changed it to not return a reference and now it's copies all around. You're probably right this is not 100% optimal though. Let me try it again in a bit.

Reply:
Sooo I read a bit on this topic and did some benchmarks and I have decided to change this function to not use `stringstream`. We're working with just strings (and a character splitter) which makes using `stringstream` not optimal, especially because `stringstream::str` creates a copy of its internal buffer, which results in an extra allocation. The case would be different if we were to print some numbers (for instance) in the resulting string, then using `stringstream` would make a lot more sense.

Now we'll be passing only references to the cache around which should be a tad more performant than the previous solution.

Reply:
looks nice!
will do one more smoke test before merging to master

Reply:
I think copying still needs to occur in this method as I'm getting a weird behavior with tags:
![image](https://user-images.githubusercontent.com/16451391/87051732-a4491000-c208-11ea-90ea-6c24a29f5c5f.png)

Reply:
one thought on how to avoid that copy - we might pass an existing std::string reference into the method for creating a tags string, and then have a separate method used by the database code that would pass a `thread_local` string into this method.

Reply:
I'll write a test for the tag splitter and joiner methods to be sure they work right. 
As for the passing a reference to the methods, that's probably too convoluted and would result in a lot of code outside of this function, this is performant enough in my opinion.

Reply:
I made a stupid mistake by not clearing the cache. It should be fine now.